### ISSUE TITLE
[Android] Installation error with RN 0.42.3

### ISSUE THREAD

**Issue by eard7** [wix/react-native-navigation#907]
### Issue Description
Hi, first of all thanks for this great navigation library.
I'm having one issue when installing the latest experimental version with react native 0.42.3.
The build fails when trying to compile.
The error i'm getting is
```
/node_modules/react-native-navigation/android/app/src/main/java/com/reactnativenavigation/react/NavigationReactGateway.java:113: error: getUseDeveloperSupport() in ReactNativeHostImpl cannot override getUseDeveloperSupport() in ReactNativeHost
        protected boolean getUseDeveloperSupport() {
                          ^
  attempting to assign weaker access privileges; was public
```

So, with a little of research, you could take a look to this [Android - ReactNativeHost getUseDeveloperSupport to public](https://github.com/facebook/react-native/pull/11329)

In other words, to fix you only need to change the line 113 in NavigationReactGateway, from protected to public. The final result should be
```
@Override
        public boolean getUseDeveloperSupport() {
            return NavigationApplication.instance.isDebug();
        }
```

### Steps to Reproduce / Code Snippets / Screenshots

Just start a new react-native project with the latest version (0.42.3), and then try to setup the latest react-native-navigation (2.0.0-experimental.256)

---
### Environment
* React Native Navigation version: 2.0.0-experimental.256
* React Native version: 0.42.3
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator/Debug

**Comment by guyca** [comment]
Thanks a lot for looking into the issue. Will fix asap
